services/userService.js

```diff
@@ -0,0 +1,64 @@
+const userDao = require('../models/userDao');
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const saltRounds = 10;
+
+const signUp = async (first_name, last_name, email, password) => {
+  const emailcheck = await userDao.duplicationEmail(email);
+  console.log(emailcheck);
+  if (emailcheck) {
+    const error = new Error('CANT USE EMAIL');
+    error.statusCode = 400;
+    throw error;
+  }
+
+  const emailValidation = new RegExp(
+    /^[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*[.]+[a-zA-Z]{2,3}$/i
+  );
+  const passwordValidation = new RegExp(
+    '^(?=.*[A-Za-z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=.{8,20})'
+  );
+  const hashedPassword = await bcrypt.hash(password, saltRounds);
+  if (!emailValidation.test(email)) {
+    const error = new Error('INVALID EMAIL');
+    error.statusCode = 400;
+    throw error;
+  }
+  if (!passwordValidation.test(password)) {
+    const error = new Error('INVALID PASSWORD');
+    error.statusCode = 400;
+    throw error;
+  }
+  const signUp = await userDao.signUp(
+    first_name,
+    last_name,
+    email,
+    hashedPassword
+  );
+  return signUp;
```

Comment:
```suggestion
  return userDao.signUp(
    first_name,
    last_name,
    email,
    hashedPassword
  );
```
변수 선언 없이 곧바로 리턴해주는 것이 깔끔해 보입니다. 또한 로직상 함수내 return 문에 작성될 코드이기에 변수 선언을 안하도록 수정한다면 await가 굳이 없어도 되겠습니다!